--- libjingle/trunk/third_party/winsdk_samples/src/Samples/multimedia/directshow/baseclasses/dllentry.cpp	Fri May 17 11:55:48 2013
+++ libjingle/trunk/third_party/winsdk_samples/src/Samples/multimedia/directshow/baseclasses/dllentry.cpp	Wed May 15 20:13:35 2013
@@ -187,7 +187,7 @@
 
 //called by COM to get the class factory object for a given class
 __control_entrypoint(DllExport) STDAPI
-DllGetClassObject(
+DllGetClassObject_X(
     __in REFCLSID rClsID,
     __in REFIID riid,
     __deref_out void **pv)
@@ -244,7 +244,7 @@
 // and CCOMObject has a static function that can tell us about the active
 // object count
 STDAPI
-DllCanUnloadNow()
+DllCanUnloadNow_X()
 {
     DbgLog((LOG_MEMORY,2,TEXT("DLLCanUnloadNow called - IsLocked = %d, Active objects = %d"),
         CClassFactory::IsLocked(),
